### PR TITLE
chore: add explainer for model introspection schema relationships

### PR DESCRIPTION
**Pull request by palpatim**
#### Checklist

- [x] PR description included

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by cwoolum — DISMISSED** [review]

File: README-relationships.md [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+        }
+```
+
+- `models.Primary` - A type definition. The **source type** for any `association`s defined in this model.
+- `models.Primary.fields.relatedMany` - The **association field**/**connection field**
+- `models.Primary.fields.relatedMany.type` - The **associated type** for this relationship. This must be a `@model`.
+- `models.Primary.fields.relatedMany.association` - The structure containing the data needed to navigate the relationship with the associated type
+- `models.Primary.fields.relatedMany.association.connectionType` - The kind of relationship (has one, has many, belongs to) this **source type** has with the associated type
+- `models.Primary.fields.relatedMany.association.associatedWith` - A list of fields on the **associated type** that hold the primary key of the **source** record. This is an array so we can support composite primary keys.
+- `models.RelatedOne` - A type definition. The **source type** for any `association`s defined in this model.
```
Comment (AaronZyLee) [inline README-relationships.md:231]: In the `Considering a snippet...` section the `models.RelatedMany` is attached but the explanation uses the `models.RelatedOne`

Comment (palpatim) [reply README-relationships.md]: D'oh! Thanks for catching. Fixed on 008fa1e

**Review by AaronZyLee — APPROVED** [review]

File: README-relationships.md [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+> **NOTE:** The MIS is an internal implementation detail of the Amplify API plugin. It should not be used in a customer application.
+
+## Sample
+
+Given a schema like
+
+```graphql
+type Primary @model @auth(rules: [{ allow: public, operations: [read] }, { allow: owner }]) {
+  id: ID! @primaryKey
+  relatedMany: [RelatedMany] @hasMany(references: "primaryId")
```
Comment (dpilch) [inline README-relationships.md:18]: Is non-array input for references supported? It is currently not supported for `fields`.

Comment (palpatim) [reply README-relationships.md:18]: Yes, it is automatically cast to an array.

**Review by dpilch — APPROVED** [review]

**Pull request merged**